### ISSUE TITLE
interested in your code

### ISSUE THREAD

**Issue by shuaiXiaoChen** [Mingyue-Cheng/CrossTimeNet#3]
你好，我对你的论文很感兴趣，然后下载你的代码运行，但在运行过程中，缺少一些文件，如在分类任务中缺少ecg-data/model.pkl，而且代码似乎默认没有预训练的过程。在预测任务中，存在同样的一个问题：vqvae_model_path = vqvae_model_prefix+datapath+'/model.pkl'这个文件也是不存在。请问一下这是我运行的方式错误，或者是什么原因？请告知指导一下，谢谢。

**Comment by Sample-design-alt** [comment]
I have the same question. Do you solve this?

**Comment by Xiaoyu-Tao** [comment]
您好，
非常感谢您对我们工作的关注，也感谢您在运行代码过程中提出的反馈。

您提到的这几个 model.pkl（例如分类任务中的 ecg-data/model.pkl，以及预测任务中的
vqvae_model_path = vqvae_model_prefix + datapath + '/model.pkl'）并不是我们预先提供的权重文件，而是需要先对时序 Tokenizer 模型进行训练后自动生成的。

具体来说：

请先进入仓库中的 Tokenizer 文件夹，运行其中的训练脚本（ main.py），对相应数据集训练时序 Tokenizer / VQ-VAE 模型；

训练完成后，程序会在对应的数据目录下生成 model.pkl（例如 ecg-data/model.pkl 或 .../model.pkl）；

完成以上步骤后，再运行分类任务或预测任务的主程序即可正常加载这些权重文件。

如果您在训练 Tokenizer 的过程中遇到具体报错或路径问题，也欢迎继续告知，我这边可以进一步帮您排查。

再次感谢！

**Comment by Sample-design-alt** [comment]
请问您可以提供对应的model.pkl吗，我想应用于Hitime，我尝试使用您提供的代码，但发现crosstimenet缺少参数，例如args里缺乏Train_data和Test_data的声明。这部分代码与Instructtime的代码是一致的吗？（但我尝试复现instructtime也失败了）
`import torch
import torch.utils.data as Data
from args import Train_data, Test_data`

**Comment by Xiaoyu-Tao** [comment]
Hi, thanks for the follow-up!

Yes — in CrossTimeNet, the pre-training initialization happens inside:

```
CrossTimeNet/forcasting/Bert4ts.py
    →  def _initialize_model(self, config):
```

This function loads the pre-trained BERT parameters and uses them to initialize the forecasting backbone. So the model is not loaded from a separate model.pkl; instead, the parameters are imported directly from the BERT checkpoint during initialization.

To reproduce CrossTimeNet:

1. You do not need model.pkl.
   The model is initialized from a pre-trained BERT inside _initialize_model().

2. Data loading should follow the DataLoader in:

```
forcasting/data_provider/
    ├── data_factory.py
    └── data_loader.py
```

This DataLoader follows the same logic as THU’s Time-Series-Library:
(https://github.com/thuml/Time-Series-Library)
It handles dataset splitting and normalization automatically, so no `Train_data` or `Test_data` arguments are required.

If you still have trouble running the code or evaluating HiTime, feel free to send me the error message — I’m happy to help debug!
